src/context.rs

```diff
@@ -280,7 +280,7 @@ impl<'a> Context<'a> {
                 let shared_repo =
                     match ThreadSafeRepository::discover_with_environment_overrides_opts(
```

Comment:
Per the docs, `discover_with_environment_overrides_opts` should already apply `GIT_CEILING_DIRECTORIES`. At a glance, it may not actually be doing this right now. Please try fixing the behavior in gitoxide first (or align the docs with the reality).

Reply:
My bad, it actually reads the environment variables [here](https://github.com/Byron/gitoxide/blob/78517a616b4b1b6fefc2367057a9b3b786d93750/git-repository/src/discover.rs#L84), so this PR effectively only adds a test.